spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/apache/RetryableRibbonLoadBalancingHttpClient.java

```diff
@@ -124,7 +126,13 @@ public RibbonApacheHttpResponse doWithRetry(RetryContext context) throws IOExcep
 				RibbonApacheHttpRequest newRequest = request;
 				if(context instanceof LoadBalancedRetryContext) {
 					ServiceInstance service = ((LoadBalancedRetryContext)context).getServiceInstance();
-					if(service != null) {
+					if (service == null) {
```

Comment:
Can we move this code into its own method, perhaps in the super class?

Reply:
yes.  I have moved the validation logic to the AbstractLoadBalancingClient.